### ISSUE TITLE
Behaviour of app after risk encounter

### ISSUE THREAD

**Issue by GiroHero256** [corona-warn-app/cwa-app-ios#1405]
## Avoid duplicates
* [X] Question is not already answered in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Question has not already been asked in another issue

## Your Question

* Source File:
* Line(s):
* Question:
The app showed a risk encounter 14 days ago. On day 13 after that encounter the app showed 12 risk encounters 2 days before. Today on day 14 the app went to green status and the 12 risk encounters became low risk encounters. Is that how this is supposed to work? This is somewhat irritating. Are all low risk encounters displayed as risk encounters if the app is in status red? Is there a way to reset the status to green after negative tests?

**Comment by daimpi** [comment]
@GiroHero256 
Which CWA and iOS version do you use? 
If you're on CWA 1.5, when did you upgrade from 1.3?

> Today on day 14 the app went to green status and the 12 risk encounters became low risk encounters. Is that how this is supposed to work?

This is indeed suspicious. It sounds similar to issue https://github.com/corona-warn-app/cwa-app-ios/issues/1365 which is currently being investigated.

>  Are all low risk encounters displayed as risk encounters if the app is in status red?

Yes: https://github.com/corona-warn-app/cwa-documentation/issues/422

> Is there a way to reset the status to green after negative tests?

No. You'll have to wait until the red exposure drops out 15 days after the actual exposure. The only other option is to delete all the keys stored in the Exposure Notification Framework (ENF) in the iOS settings. But in this case you will loose all the keys you sent and received in the last 14 days, which I would advise against.

**Comment by GiroHero256** [comment]
Im on iOS 14.1, The app shows version 1.5.1 (1).

**Comment by TPei** [comment]
Hey there, similar issue here. Showed "3 exposures" for "2 days ago" on Saturday morning, claiming "increased risk". This morning I get a notification and it now says "3 encounters with low risk". I'm also running 1.5.1 (1) on iOS 14.0.1. The app shows as having been updated 3 days ago. Sadly it's unclear if that update was before or after my notification on Saturday morning. On Sunday however it still showed as increased risk with 3 exposures.

**Comment by ndegendogo** [comment]
@GiroHero256 Could you export your ENF log file (json)and post it?
Alternatively, you can analyse it on your own.
Under iOS 14.1 it shows the number of matches per file, and it shows the history of this matching procedure. **But it does not distinguish between red and green matches.**
Note: the individual data in your json ENF log file are the matches, and your device type and iOS version. The key files (hash values and key count) are public, and everybody here gets the same.

**Comment by GiroHero256** [comment]
[ExposureChecks-2020-10-26.zip](https://github.com/corona-warn-app/cwa-app-ios/files/5437930/ExposureChecks-2020-10-26.zip)

**Comment by ndegendogo** [comment]
@TPei I recommend that you update your iOS to 14.1. 
Unfortunately, iOS 14.0.1 does not show number of matches in the ENF log file (see #1106 for more details), so it is hard to analyse this in retrospective. Still some degree of information could be in your ENF log after your next check under iOS 14.1

**Comment by ndegendogo** [comment]
@GiroHero256 Analysing the history of your checks (latest entry is on top of the json file).
1) on 25 Oct there is:
Hash 106809... with 1 match
Hash 064B72... with 10 matches
Hash D6C9AD ... with 1 match
all other with 0 matches -> total 12
Combined with your cwa info of that day, all these are green
2) on 24 Oct we have the same 10 + 1 (Hash 06... and D6...) (green), but then another
Hash 8F2EA0... with 1 match.
-> total again 12. 
Combined with your cwa info of that day, this one (Hash 8F...) was red.
3) Reason why this red match does not show up on 25 Oct is probably because the related RPI was deleted after its lifetime of 14 days
4) We can also follow this red one into the past, we see it every day, first on 22 Oct.
**On 21 Oct we see this same file checked, but no match logged.** I assume this was an older iOS version?

So, to summarize:
- cwa shows the risk level (red or green) based on the maximum risk of all encounters checked
- cwa shows the total number of encounters, but does not tell you how many of them were red or green
- the same day on which your red encounter faded out, you got a new match, but this one was green

**Comment by GiroHero256** [comment]
Thanks for the explanation.

"On 21 Oct we see this same file checked, but no match logged."
I assume that I updated the App from 1.5 to 1.5.1 on 21.10..

I would still like to ask to consider changing the behaviour as suggested by others e.g. in corona-warn-app/cwa-documentation#422 or  #1408.

I was irritated by getting new risk exposures which turned out to be low risk. It might lead to unnecessary testing or - if the warning would be ignored - not reacting properly to actual risk exposures.

**Comment by GiroHero256** [comment]
I consider the question answered. There are other issues for the change I suggested.

**Comment by ndegendogo** [comment]
@GiroHero256 actually, I am not sure if the notion of "red" (high-risk) and "green" (low-risk) encounters is 100% correct, because they are not considered in isolation. Rather all these encounters are summed up, and contribute a large or a small amount to the overall risk

**Comment by daimpi** [comment]
@ndegendogo that's actually a very good and subtle point. There are some green contacts which will contribute nothing because their "[total risk](https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#risk-score-calculation)" is smaller than 11 (i.e. smaller than the `minimalRiskScore`) but there will be others which combine together to trigger the "red" status even though each on their own would not suffice, exactly as in the [example](https://covidapps.slack.com/archives/C0194ML0MLN/p1602366095339900?thread_ts=1602134945.259900&cid=C0194ML0MLN) we recently discussed on  the [community Slack](https://github.com/corona-warn-app/cwa-documentation/issues/379) 🙂.
